public/app/features/explore/QueryRow.tsx

```diff
@@ -102,12 +100,13 @@ export class QueryRow extends PureComponent<QueryRowProps, QueryRowState> {
   };
 
   setReactQueryEditor = () => {
-    const { mode, datasourceInstance } = this.props;
+    const { datasourceInstance } = this.props;
     let QueryEditor;
 
-    if (mode === ExploreMode.Metrics && datasourceInstance.components?.ExploreMetricsQueryField) {
+    // TODO:unification
```

Comment:
Looks like there is no other way than removing this field specialisation hand add a switcher to the field itself.